pkg/server/server.go

```diff
@@ -141,6 +142,7 @@ type Server struct {
 	dispatchThrottlingCheckResolverFrequency time.Duration
 	dispatchThrottlingThreshold              uint32
 
+	dispatchThrottler               *throttler.DispatchThrottler
```

Comment:
Had this comment 1:1, but reposting here for broader visibility.

My only reservation right now with these changes is that ListObjects and Check will share a dispatch throttler, and so that throttling queue is going to have a lot contention potentially. I’m not sure if that’s the right approach, and I’d like to understand better if they should share or have independent throttlers. My hunch is independent throttlers (at least to start).

Can we analyze and better understand this before we consolidate them into a shared throttler?